staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -102,6 +102,12 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx context.Context, obj runtime.
 		objectMeta.SetClusterName("")
 	}
 
+	if mutateObjectFunc, ok := MutateObjectFuncFrom(ctx); ok {
+		if err := mutateObjectFunc(ctx, obj, nil); err != nil {
+			return err
+		}
+	}
+
```

Comment:
Moving this here allows admission to set fields that are required to be set consistently by the strategy (in PrepareForCreate) or by the API server (SetDeletionTimestamp, SetDeletionGracePeriodSeconds, FillObjectMetaSystemFields, etc).

That means admission can violate API invariants. A few examples:
* could set status fields on resource create
* could set metadata.uid to non-unique values
* could set deletion timestamp on create
* could set fields that are not permitted, like metadata.clusterName

Additionally, I think this is after metadata.managedFields are inspected/set, which is not where I would expect admission to be making changes.